### ISSUE TITLE
Difficulties for installation

### ISSUE THREAD

**Issue by Rob192** [automl/auto-sklearn#1251]
## Describe the bug ##
Hello ! thanks for auto-sklearn ! it looks great :)
However, I am experimenting some issues when installing the package and I hoped you could help. I am having several issues : 
1. On my server (linux, venv), the installation takes a while when using `pip install auto-sklearn`. I am seeing many message saying `INFO: pip is looking at multiple versions of scikit-learn to determine which version is compatible with other requirements. This could take a while.` for all the packages and it takes for ever to install. At the end, I am seeing `INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. If you want to abort this run, you can press Ctrl + C to do so. To improve how pip performs, tell us what happened here: https://pip.pypa.io/surveys/backtracking`. Any idea how to speed up the process ? 
2. On a docker on my local computer (I am running Windows so no choice !), the version I am installing is `0.12.7` and not the last one `0.14.0`. Therefore, the Multi-label Classification exemple does not work. 

## To Reproduce ##
This is my dockerfile: 
```Dockerfile
FROM tensorflow/tensorflow:latest-gpu

COPY requirements.txt requirements.txt

RUN apt-get update \
    && apt-get install --no-install-recommends ffmpeg libsm6 libxext6 poppler-utils -y \
    && apt-get install --no-install-recommends build-essential swig python3-dev -y \
    && pip install --upgrade pip setuptools wheel \
    && pip install -r requirements.txt \
    && pip cache purge \
    && apt-get autoremove -y \
    && rm -rf /var/lib/apt/lists/* \
    && rm -rf /root/.cache/pip

RUN python -m spacy download fr_core_news_lg

COPY download_doctr_models.py .
RUN python download_doctr_models.py

COPY . .
WORKDIR .
CMD [ "streamlit", "run", "app_local.py", "--server.enableCORS=false", "--server.enableXsrfProtection=false","--server.enableWebsocketCompression=false" ]
```

## Expected behavior ##
0.14.0 should install super fast :D

**Comment by eddiebergman** [comment]
Hi @Rob192,

### `0.12.7`
On the note about `0.12.7`, we released the version as is and unfortunately that failed test should have been flagged in out tests. We are unlikely going to go back and fix this as we are still in `0.x.x` versioning. Thanks for letting us know though!

### `pip` problems
I just did a test on my on local machine and did not experience the same issues, everything installed without an issue and in a reasonable time (~20s).
```bash
## Installing from pip
# --no-cache-dir forces pip to not use any cached libraries

$ ..make venv
$ pip install --upgrade pip
$ pip install --no-cache-dir auto-sklearn 

## Install from source with requirements.txt
$ ..make venv
$ git clone git@github.com:automl/auto-sklearn.git
$ cd auto-sklearn
$ pip install --upgrade pip
$ pip install --no-cache-dir -r requirements.txt
```

My only guess would be that other libraries you use outside of `auto-sklearn` also require `scikit-learn` but a different version that may be incompatible. Even then, I would expect pip to fail in that case. I'm afraid I've never seen this issue before. 

As a temporary fix, if critical, you could explicitly state which version of `scikit-learn` to install, choosing the one that `pip` eventually settles on.

```bash
# For example, with no other libraries but auto-sklearn
$ pip list | grep scikit
scikit-learn      0.24.2
```

Could I ask you to provide the following information from whatever environment you are installing from?
```bash
$ uname -a
Linux myhost 5.14.5-arch1-1 #1 SMP PREEMPT Thu, 16 Sep 2021 11:02:40 +0000 x86_64 GNU/Linux
$ python -V
Python 3.8.6
$ pip -V
pip 21.2.4 from /path/to/venv/path/to/pip (python 3.8)
```

**Comment by Rob192** [comment]
Hello,

Some feedback after some more trials :
- Regarding version `0.12.7` : Now it seems that everything is running as expected : when I install via pip I get the correct version number `0.14.0` and the script for the Multi-label Classification exemple is working as expected. I think my mistake was coming from the fact that I used a Docker image of Tensorflow that uses python 3.6 -> SOLVED
- Regarding the pip install : I started a new virtual environment and installed everything from scratch and it worked fine. 

Thanks for your answer anyways

**Comment by eddiebergman** [comment]
> Hello,
> 
> Some feedback after some more trials :
> 
>     * Regarding version `0.12.7` : Now it seems that everything is running as expected : when I install via pip I get the correct version number `0.14.0` and the script for the Multi-label Classification exemple is working as expected. I think my mistake was coming from the fact that I used a Docker image of Tensorflow that uses python 3.6 -> SOLVED
> 
>     * Regarding the pip install : I started a new virtual environment and installed everything from scratch and it worked fine.
> 
> 
> Thanks for your answer anyways

Hi @Rob192,

Glad it worked out for you and thanks for the feedback. We don't officially support 3.6 and it should fail during installation but perhaps we should put in an explicit check in the source code itself.

Best!
